Sources_v3/Workers/MessageUpdater.swift

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+class MessageUpdater<ExtraData: ExtraDataTypes>: Worker {
+    func getMessage(cid: ChannelId, messageId: MessageId, completion: ((Error?) -> Void)? = nil) {
+        let endpoint: Endpoint<MessagePayload<ExtraData>> = .getMessage(messageId: messageId)
+        apiClient.request(endpoint: endpoint) {
+            switch $0 {
+            case let .success(message):
+                self.database.write({ session in
+                    try session.saveMessage(payload: message, for: cid)
+                }, completion: { error in
+                    completion?(error)
+                })
+            case let .failure(error):
+                completion?(error)
+            }
+        }
+    }
+    
+    func deleteMessage(messageId: MessageId, completion: ((Error?) -> Void)? = nil) {
+        var shouldDeleteOnBackend = true
+        
+        database.write({ session in
+            guard
+                let dto = session.message(id: messageId), [.pendingSend, .sendingFailed].contains(dto.localMessageState)
```

Comment:
Should we also throw `ClientError.MessageDoesNotExist` when the message doesn't exist?

Reply:
I don't think so. We can have a message that exists on the backend but hasn't been fetched yet (we discussed that we don't require `startUpdating` to be called before accessing `deleteMessage`)